oauth2_http/java/com/google/auth/oauth2/AwsCredentialSource.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.auth.oauth2;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The AWS credential source. Stores data required to retrieve the AWS credential from the AWS
```

Comment:
Also: https://github.com/googleapis/google-auth-library-java/actions/runs/5590404249/job/15141929998?pr=1236 @TimurSadykov

Reply:
I think that's because I said to rename the base class, we called it CredentialSource previously, when it should be ExternalAccountCredentialSource which matters more now that they are public. The errors is a bit weird since the builder is public even though no one would have been able to instantiate the credentialSource object in the first place, so not sure if this is really breaking

Reply:
We also have a getCredentialSource method that is public for some reason.. but I think you're right if we don't rename it we're good

Reply:
If I understand correctly its not only about name, rather the fact that base class has to stay nested. Otherwise imports has to be updated and therefore the change is breaking. 

I think it is ok if we keep the base class nested until the major bump, but move child classes as separate.

Reply:
CredentialSource is package private, shouldn't be an issue?